legal-api/src/legal_api/reports/report_v2.py

```diff
@@ -0,0 +1,242 @@
+            # substitute template parts
+            template_code = self._substitute_template_parts(template_code)
+        except Exception as err:
+            current_app.logger.error(err)
+            raise err
+        return template_code
+
+    @staticmethod
+    def _substitute_template_parts(template_code):
+        template_path = current_app.config.get('REPORT_TEMPLATE_PATH')
+        template_parts = [
+            'common/v2/style',
+            'common/v2/styleMail',
+            'common/certificateRegistrarSignature'
+        ]
+        # substitute template parts - marked up by [[filename]]
+        for template_part in template_parts:
+            template_part_code = Path(f'{template_path}/template-parts/{template_part}.html')\
+                .read_text(encoding='UTF-8')
+            template_code = template_code.replace('[[{}.html]]'.format(template_part), template_part_code)
+        return template_code
+
+    def _get_template_data(self):
+        self._report_data = {}
+        self._format_furnishing_data()
+        self._set_meta_info()
+        self._set_address()
+        self._set_registrar_info()
+        if self._document_key == ReportTypes.DISSOLUTION:
+            self._set_ep_registration()
+        return self._report_data
+
+    def _format_furnishing_data(self):
+        self._report_data['furnishing'] = {
+            'businessName': self._furnishing.business_name,
+            'businessIdentifier': self._furnishing.business_identifier
+        }
+
+        if self._furnishing.last_ar_date:
+            last_ar_date = LegislationDatetime.as_legislation_timezone(self._furnishing.last_ar_date)
+        else:
+            last_ar_date = LegislationDatetime.as_legislation_timezone(self._business.founding_date)
+        self._report_data['furnishing']['lastARDate'] = last_ar_date.strftime(OUTPUT_DATE_FORMAT)
+
+        if self._furnishing.processed_date:
+            processed_date = LegislationDatetime.as_legislation_timezone(self._furnishing.processed_date)
+        else:
+            processed_date = LegislationDatetime.as_legislation_timezone(self._report_date_time)
+        self._report_data['furnishing']['processedDate'] = processed_date.strftime(OUTPUT_DATE_FORMAT)
+
+    def _set_meta_info(self):
+        if self._output_type:
+            self._report_data['outputType'] = self._output_type
+        else:
+            self._report_data['outputType'] = 'email'
+        self._report_data['title'] = ReportMeta.reports[self._document_key]['reportDescription'].upper()
+
+    def _set_address(self):
+        if (furnishing_address := Address.find_by(furnishings_id=self._furnishing.id)):
+            furnishing_address = furnishing_address[0]
+            self._report_data['furnishing']['mailingAddress'] = furnishing_address.json
+        elif (mailing_address := self._business.mailing_address.one_or_none()):
+            self._report_data['furnishing']['mailingAddress'] = mailing_address.json
```

Comment:
For now I think we should remove this.

report v2 is only used for furnishing entry related outputs right now and if an address should always be tied to a furnishing entry.  If it isn't, it's bad data and which should be data fixed.